spec/features/audit/report/csv_export_spec.rb

```diff
@@ -13,99 +7,140 @@ def content_disposition
-        link_type: "primary_publishing_organisation",
-      )
-
-      create(
-        :content_item,
-        title: "Example 2",
-        base_path: "/example2",
-      )
-
-      create(:audit, content_item: example1)
-    end
+  scenario "Exporting a csv file as an attachment" do
+    given_i_am_an_auditor_belonging_to_an_organisation
+    when_i_export_an_audit_report
+    then_i_receive_the_report_in_csv_format
+  end
 
-    scenario "Exporting a csv file as an attachment" do
-      visit audits_report_path
-      click_link "Export filtered audit to CSV"
+  scenario "Applying the filters to the export" do
+    given_i_am_an_auditor_belonging_to_an_organisation
+    and_filter_is_applied_to_show_only_hmrc_content_items
+    when_i_export_the_reports
+    then_i_receive_only_hmrc_related_content_items_in_the_report
+  end
 
-      expect(content_type).to eq("text/csv")
-      expect(content_disposition).to start_with("attachment")
-      expect(content_disposition).to include(
-        'filename="Transformation_audit_report_CSV_download.csv"',
-      )
+  scenario "Multiple pages including_details_of_audit_progress_ of content items are in the database" do
+    given_i_am_an_auditor_belonging_to_an_organisation
+    and_there_are_multiple_pages_of_unfiltered_content_items
+    when_i_export_all_the_reports
+    then_i_receive_an_audit_progress_report_for_all_content_items_in_csv_format
+  end
 
-      expect(page).to have_content("Title,URL")
-      expect(page).to have_content("Example 1,https://gov.uk/example1")
-    end
+  scenario "Discard audit status filter when clicking from content view to report, and then exporting CSV" do
+    given_i_am_an_auditor_belonging_to_an_organisation
+    and_there_are_two_unassigned_content_items
+    when_i_filter_the_content_page_for_content_items_with_no_one_assigned_to_them
+    then_i_see_the_two_unassigned_content_items
+    when_i_navigate_to_audits_report_page
+    then_i_do_not_see_the_two_unassigned_content_items
+    and_i_see_the_one_content_item_assigned_to_me
+  end
 
-    scenario "Applying the filters to the export" do
-      visit audits_report_path
-      select "Anyone", from: "allocated_to"
+  def given_i_am_an_auditor_belonging_to_an_organisation
+    user = create(:user)
+    hmrc = create(
+      :content_item,
+      title: "HMRC",
+      document_type: "organisation",
+      allocated_to: user
+    )
+    example1 = create(:content_item,
+                      title: "Example 1",
+                      base_path: "/example1",
+                      allocated_to: user)
+    create(:link,
+           source_content_id: example1.content_id,
+           target_content_id: hmrc.content_id,
+           link_type: "primary_publishing_organisation")
+    create(:audit, content_item: example1)
+  end
 
-      select "HMRC", from: "Organisations"
-      click_on "Apply filters"
+  def then_i_receive_only_hmrc_related_content_items_in_the_report
```

Comment:
It might be nicer to name your examples a bit more descriptively:

```ruby
def then_i_receive_onlky_hmrc_related_content_items_in_the_report
  ...
  expect(@audit_report).to have_content("HMRC content...")
  expect(@audit_report).to have_no_content("Non-HMRC content...")
end
```

Reply:
Good idea! 👍